app/src/main/java/org/fossasia/susi/ai/skills/SkillsActivity.kt

```diff
@@ -72,14 +72,19 @@ class SkillsActivity : AppCompatActivity(), SkillFragmentCallback {
 
     override fun onBackPressed() {
         overridePendingTransition(R.anim.trans_right_in, R.anim.trans_right_out)
-        if (supportFragmentManager.popBackStackImmediate(TAG_SKILLS_FRAGMENT, 0)) {
-            title = getString(R.string.skills_activity)
-        } else if (supportFragmentManager.popBackStackImmediate(TAG_GROUP_WISE_SKILLS_FRAGMENT, 0)) {
-            title = getString(R.string.skills_activity)
-        } else {
+
+        var count:Int=supportFragmentManager.backStackEntryCount
+        supportFragmentManager.popBackStack()
+
+        if(count==1){
             finish()
             exitActivity(this)
-        }
+        }else if(count==2){
+            title=getString(R.string.skills_activity)
+        }else
+            title=getString(R.string.skills_activity)
+
+
```

Comment:
Not a good logic

Reply:
Thanks for your review sir.
I will try to make it better, If you have any suggestions for me then please tell me.